sources/packages/api/src/esdc-integration/fed-restriction-integration/fed-restriction-files/fed-restriction-file-record.ts

```diff
@@ -0,0 +1,61 @@
+import { getDateOnlyFromFormat } from "../../../utilities";
+import { DATE_FORMAT } from "../fed-restriction-integration.models";
+
+/**
+ * Federal restriction record that contains the restriction code and restriction reason
+ * associated with a Student.
+ * Federal restrictions on the database contains only one code that is the result of the
+ * combination of the 'restriction code' with the 'restriction reason code'(reason code is optional).
+ * The documentation about it is available on the document
+ * 'SLP-AppendixF2AsReviewed2016-FileLayouts BC Files V3(HAJ-CB EDITS) In ESDC Folder'.
+ */
+export class FedRestrictionFileRecord {
+  constructor(
+    private readonly line: string,
+    public readonly lineNumber: number,
+  ) {}
+
+  public get sin(): string {
+    return this.line.substr(0, 9);
+  }
+
+  public get surname(): string {
+    return this.line.substr(9, 30).trim();
+  }
+
+  public get dateOfBirth(): Date {
+    return getDateOnlyFromFormat(this.line.substr(69, 8), DATE_FORMAT);
+  }
+
+  public get restrictionCode(): string {
+    return this.line.substr(84, 1);
+  }
+
+  public get restrictionReasonCode(): string {
+    return this.line.substr(85, 1);
+  }
+
+  public getComposedCode() {
+    return `${this.restrictionCode}${this.restrictionReasonCode}`.trim();
+  }
+
+  /**
+   * Assess the record to know in advance if some invalid data is present.
+   * @returns invalid data message, if any.
+   */
+  public getInvalidDataMessage(): string | null {
+    const errors: string[] = [];
+    if (isNaN(+this.sin)) {
```

Comment:
if the `this.sin` is empty or `"           "`, it is considered as a valid sin, right, same in the below cases. or am I missing anything? or are we considering all these values are mandatory fields
![image](https://user-images.githubusercontent.com/77353155/144668525-0c6e40f3-2d2b-4621-9a1d-4b12783dabad.png)

Reply:
Your test is missing one part. It should be as below
```
console.log(isNaN(parseInt("   ")))
```